### PR TITLE
Memoize last shape, skip tuneBatchedGemm if not needed

### PR DESCRIPTION
**Pull request by ryanleary**
This is a workaround where sometimes configurePlugin is called unnecessarily. A better WAR might be to use a LRU or circular buffer of shape sizes to algo choices.

### CODE REVIEW

**Comment by ryanleary** [comment]
attn: @rajeevsrao

File: plugin/bertQKVToContextPlugin/qkvToContextPlugin.cu [hunk 1]
```diff
@@ -590,8 +594,12 @@ void QKVToContextPluginDynamic::configurePlugin(
 
     const int S = in->max.d[SDIM];
     const int B = in->max.d[BDIM];
-    std::tie(mAlgoBatchedEx1, mAlgoBatchedEx2) = tuneBatchedGemm(B, S, mNumHeads, mHeadSize);
-    gLogVerbose << "QKV Plugin - Selected Algos for batch gemms: " << mAlgoBatchedEx1 << ", " << mAlgoBatchedEx2 << "\n";
+    if ((S != mS) || (B != mB)) {
```
Comment (rajeevsrao) [inline plugin/bertQKVToContextPlugin/qkvToContextPlugin.cu:597]: @ryanleary also need to check if `in->max.d[SDIM/BDIM]` was changed, which also takes the if case.

Comment (ryanleary) [reply plugin/bertQKVToContextPlugin/qkvToContextPlugin.cu:597]: Can you clarify what extra check is needed here? I'm not following. `SDIM = 0` so `in->max.d[SDIM] == in->max.d[SDIM/BDIM]`.

**Comment by ttyio** [comment]
When dynamic shape is used, we should tune the batch gemm again when user binding shape changed, this happen in `configurePlugin` instead of the constructor, the latest code only tune the perf again when either B or S changed. I will close this one, thank you for your contribution!

**Pull request closed without merging**